config/NaoH25V40.srdf

```diff
@@ -3,88 +3,112 @@
-	<chain base_link="torso" tip_link="r_wrist" />
+        <chain base_link="torso" tip_link="r_wrist" />
+    </group>
+    <group name="right_hand">
+        <link name="RFinger11_link" />
+        <link name="RFinger12_link" />
+        <link name="RFinger13_link" />
+        <link name="RFinger21_link" />
+        <link name="RFinger22_link" />
+        <link name="RFinger23_link" />
+        <link name="r_gripper" />
+        <link name="RHandTouchBack_frame" />
+        <link name="RHandTouchLeft_frame" />
+        <link name="RHandTouchRight_frame" />
+        <link name="RThumb1_link" />
+        <link name="RThumb2_link" />
+#        <link name="r_wrist" />
+        <joint name="RHand" />
+#        <joint name="RWristYaw" />
+    </group>
+    <group name="base">
+        <joint name="virtual_joint" />
     </group>
-
     <!--END EFFECTOR: Purpose: Represent information about an end effector.-->
-<!--    <end_effector name="left_eef" parent_link="l_gripper" group="LeftArmWithHand" />
-    <end_effector name="right_eef" parent_link="r_gripper" group="RightArmWithHand" /> -->
-<!--    <end_effector name="leftarm_eef" parent_link="LForeArm" group="LeftArm" />
-    <end_effector name="rightarm_eef" parent_link="RForeArm" group="RightArm" /> -->
-<!--    <end_effector name="left_eef" parent_link="l_wrist" group="LeftArmWithHand" />
-    <end_effector name="right_eef" parent_link="r_wrist" group="RightArmWithHand" />
--->    <!--VIRTUAL JOINT: Purpose: this element defines a virtual joint between a robot link and an external frame of reference (considered fixed with respect to the robot)-->
-    <virtual_joint name="virtual_joint" type="fixed" parent_frame="odom" child_link="base_link" />
+    <end_effector name="left_eef" parent_link="l_wrist" group="left_hand" />
+    <end_effector name="right_eef" parent_link="r_wrist" group="right_hand" />
+    <!--VIRTUAL JOINT: Purpose: this element defines a virtual joint between a robot link and an external frame of reference (considered fixed with respect to the robot)-->
+    <virtual_joint name="virtual_joint" type="planar" parent_frame="odom" child_link="base_link" />
     <!--PASSIVE JOINT: Purpose: this element is used to mark joints that are not actuated-->
-    <passive_joint name="LFinger11" />
-    <passive_joint name="LFinger12" />
-    <passive_joint name="LFinger13" />
-    <passive_joint name="LFinger21" />
-    <passive_joint name="LFinger22" />
-    <passive_joint name="LFinger23" />
-    <passive_joint name="LThumb1" />
-    <passive_joint name="LThumb2" />
-    <passive_joint name="RHipYawPitch" />
-    <passive_joint name="RFinger11" />
-    <passive_joint name="RFinger12" />
-    <passive_joint name="RFinger13" />
-    <passive_joint name="RFinger21" />
-    <passive_joint name="RFinger22" />
-    <passive_joint name="RFinger23" />
-    <passive_joint name="RThumb1" />
-    <passive_joint name="RThumb2" />
+    <passive_joint name="virtual_joint" />
```

Comment:
I think that the virtual joint is known to be virtual. I think you don't have to put it as passive joint.